pkg/controller/hibernation/hibernation_controller.go

```diff
@@ -413,6 +417,21 @@ func (r *hibernationReconciler) checkClusterResumed(cd *hivev1.ClusterDeployment
 		logger.WithError(err).Log(controllerutils.LogLevel(err), "Failed to connect to target cluster")
 		return reconcile.Result{}, err
 	}
+
+	preemptibleActuator, ok := actuator.(HibernationPreemptibleMachines)
+	if ok {
+		replaced, err := preemptibleActuator.ReplaceMachines(cd, remoteClient, logger)
+		if err != nil {
+			logger.WithError(err).Log(controllerutils.LogLevel(err), "Failed to replace Preemptible machines")
+			return reconcile.Result{}, err
+		}
+		if replaced {
+			// when machines were replaced we must give some time before new nodes
+			// appear.
+			return reconcile.Result{RequeueAfter: nodeCheckWaitTime}, nil
+		}
```

Comment:
Making sure I'm understanding the logic here:

As written, the err will cause an immediate requeue, even if *some* machines were replaced. So the subsequent reconcile will find the as-yet-unreplaced machines and try to replace them. Eventually we'll get to a point where the last errant machine has been replaced and there is no error. So it's at that point that we'll wait `nodeCheckWaitTime` to requeue.

Or, shorter: we'll wait `nodeCheckWaitTime` to requeue when we've successfully replaced the *last* machine.

Reply:
> Or, shorter: we'll wait nodeCheckWaitTime to requeue when we've successfully replaced the last machine.

wait after replacing **any** machine. This is allow for new machine is boot in replacement and join as a node.

Reply:
As written, it won't wait if there's any error.

Reply:
which is correct. failing on error moves the work to rate-limited queue. so transient errors are retired and given up (very large next interval) on continued failures.